### PR TITLE
gadget: add writer for offset-write

### PR DESCRIPTION
**Pull request by bboozzoo**
Add writer for deploying offset-write entries to the image

### CODE REVIEW

File: gadget/offset.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+func (w *OffsetWriter) Write(out io.WriteSeeker) error {
+	// positioning guarantees that start offset is aligned to sector size
+
+	if w.ps.PositionedOffsetWrite != nil {
+		if err := deployOffsetWrite(out, *w.ps.PositionedOffsetWrite, asLBA(w.ps.StartOffset, w.sectorSize)); err != nil {
+			return err
+		}
+	}
+
+	if !w.ps.IsBare() {
```
Comment (stolowski) [inline gadget/offset.go]: I think this case is tested by giving and non-giving Filesystem in the tests inputs, is that right?

File: gadget/offset.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		// only raw content uses offset-writes
+		return nil
+	}
+
+	for _, pc := range w.ps.PositionedContent {
+		if pc.PositionedOffsetWrite == nil {
```
Comment (stolowski) [inline gadget/offset.go]: Is that case tested?

File: gadget/offset.go (hunk 1 above)
Comment (bboozzoo) [reply gadget/offset.go]: Yup

File: gadget/offset.go (hunk 2 above)
Comment (bboozzoo) [reply gadget/offset.go]: Yes, see https://github.com/snapcore/snapd/pull/7003/commits/ece80c25f1039e8c3d640d0664c3ceddcaa9be56#diff-341e2c070839d7b2c0b6c99cc764f8d7R45 structure declares offset-write but its content does not.

File: gadget/offset.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+type OffsetWriter struct {
+	ps         *PositionedStructure
+	sectorSize Size
+}
+
+func asLBA(value, sectorSize Size) uint32 {
+	return uint32(value / sectorSize)
+}
+
+func deployOffsetWrite(out io.WriteSeeker, offset Size, value uint32) error {
```
Comment (zyga) [inline gadget/offset.go]: Nitpick: in the kernel things like that are called `pwrite`, perhaps we could mimic the name?

**Pull request merged**